common/onion.c

```diff
@@ -176,10 +177,15 @@ static bool pull_payload_length(const u8 **cursor,
 	 *   case the `hop_payload_length` is equal to the numeric value of
 	 *   `length`.
 	 */
-	if (*len > 1) {
+	if (!has_realm || *len > 1) {
```

Comment:
what if `*len == 1`?
```suggestion
	if (!has_realm || *len > 0) {
```

Reply:
oh i just saw this comment ... ignore this.

```
 identified by any length over **`1`**
```